Refactor Eve build description

gdemonet opened the pull request:
**Component**: ci

**Context**: The existing `eve/main.yml` was littered with repetitions, making adjustments to certain steps cumbersome.

**Summary**:

- Refactor of the whole file using YAML anchors, references, and extends (see [this blog post](https://blog.daemonl.com/2016/02/yaml.html) for more info)
- Fine-tune timeout durations for artifacts retrieval
- Add some retry logic for retrieving large artifacts (sometimes, a `curl` request will fail with an empty response)

**Acceptance criteria**: 

- Builds still pass
- Less flakiness on artifacts retrieval

---

**Note**: this is a temporary refactoring - we plan on writing a custom "DSL" for generating this file at build-time, instead of these complex YAML tricks.

**To do**:

- [x] Rebase on `development/2.0` and adjust upgrade/downgrade specificities in according branches

gdemonet committed:
ci: Refactor Eve YAML description

Our `eve/main.yml` build description was becoming too repetitive,
involving numerous subtle differences between similar stages (e.g.
between `single-node-upgrade` and `single-node-downgrade`)

A better solution in the future will be to generate this YAML file from
a "real" programming language, using some homemade DSL.

gdemonet committed:
ci: Bump timeout for retrieving ISO from artifacts

We tend to hit 20min timeouts when retrieving ISO images from artifacts,
so we bump the maximum value to avoid failing in case the artifacts
service is too slow.

This commit can be reverted once the artifacts service gets upgraded to
something more "punchy".

gdemonet committed:
ci: Lower timeout for downloading SHA256 checksum

Downloading this small file shouldn't take longer than a few minutes, so
we enforce a smaller timeout on this command.

gdemonet committed:
ci: Add retrying logic for large artifacts

Sometimes, it seems the artifacts service returns an empty content,
yielding a returncode 18 from `curl`. We retry the call until we hit a
maximum number of retries, to alleviate such situations.

bert-e commented:
# Hello gdemonet,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

bert-e commented:
# Branches have diverged



This pull request's source branch `bugfix/increase-timeout-for-large-artifacts` has diverged from
`development/2.4` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.4` into `bugfix/increase-timeout-for-large-artifacts`
* Rebase `bugfix/increase-timeout-for-large-artifacts` onto `origin/development/2.4`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.1/bugfix/increase-timeout-for-large-artifacts` with contents from `bugfix/increase-timeout-for-large-artifacts`
and `development/2.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.1/bugfix/increase-timeout-for-large-artifacts origin/development/2.1
 $ git merge origin/bugfix/increase-timeout-for-large-artifacts
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.1/bugfix/increase-timeout-for-large-artifacts
```

gdemonet commented:
/status

bert-e commented:
# Status





*Status report is not available.*

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.3/bugfix/increase-timeout-for-large-artifacts` with contents from `w/2.2/bugfix/increase-timeout-for-large-artifacts`
and `development/2.3`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.3/bugfix/increase-timeout-for-large-artifacts origin/development/2.3
 $ git merge origin/w/2.2/bugfix/increase-timeout-for-large-artifacts
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.3/bugfix/increase-timeout-for-large-artifacts
```

gdemonet commented:
/status

bert-e commented:
# Status





*Status report is not available.*

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.4/bugfix/increase-timeout-for-large-artifacts` with contents from `w/2.3/bugfix/increase-timeout-for-large-artifacts`
and `development/2.4`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.4/bugfix/increase-timeout-for-large-artifacts origin/development/2.4
 $ git merge origin/w/2.3/bugfix/increase-timeout-for-large-artifacts
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.4/bugfix/increase-timeout-for-large-artifacts
```

gdemonet commented:
/create_pull_requests

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/increase-timeout-for-large-artifacts` into
`development/2.0`
* [w/2.1/bugfix/increase-timeout-for-large-artifacts](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/bugfix/increase-timeout-for-large-artifacts) will be merged into `development/2.1` (pull request #1953)
* [w/2.2/bugfix/increase-timeout-for-large-artifacts](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/bugfix/increase-timeout-for-large-artifacts) will be merged into `development/2.2` (pull request #1954)
* [w/2.3/bugfix/increase-timeout-for-large-artifacts](https://github.com/scality/metalk8s/compare/development/2.3...w/2.3/bugfix/increase-timeout-for-large-artifacts) will be merged into `development/2.3` (pull request #1955)
* [w/2.4/bugfix/increase-timeout-for-large-artifacts](https://github.com/scality/metalk8s/compare/development/2.4...w/2.4/bugfix/increase-timeout-for-large-artifacts) will be merged into `development/2.4` (pull request #1956)



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **create_pull_requests**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #46226d26f83e8184255ca8842bf93f49b96c1961 on the integration branch
`w/2.2/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.2`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #46226d26f83e8184255ca8842bf93f49b96c1961 on the integration branch
`w/2.3/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.3`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #46226d26f83e8184255ca8842bf93f49b96c1961 on the integration branch
`w/2.4/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.4`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #c4b27553d517b08e0fe3577bd8a780156153d6bf on the integration branch
`w/2.2/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.2`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #2aa0884a0cf6abc702a24cb24e566cc3972574d6 on the integration branch
`w/2.3/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.3`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #46226d26f83e8184255ca8842bf93f49b96c1961 on the integration branch
`w/2.4/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.4`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **create_pull_requests**

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/2.1/bugfix/increase-timeout-for-large-artifacts` with contents from `bugfix/increase-timeout-for-large-artifacts`
and `development/2.1`.

Please resolve the conflict on **the integration branch** (`w/2.1/bugfix/increase-timeout-for-large-artifacts`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/2.1/bugfix/increase-timeout-for-large-artifacts
 $ git pull  # or "git reset --hard origin/w/2.1/bugfix/increase-timeout-for-large-artifacts"
 $ git merge origin/development/2.1
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/bugfix/increase-timeout-for-large-artifacts
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.1/bugfix/increase-timeout-for-large-artifacts
```





*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #3ed4e2056028b492433ea346419c9c3174c05582 on the integration branch
`w/2.1/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #3ed4e2056028b492433ea346419c9c3174c05582 on the integration branch
`w/2.2/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.2`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #3ed4e2056028b492433ea346419c9c3174c05582 on the integration branch
`w/2.3/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.3`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #3ed4e2056028b492433ea346419c9c3174c05582 on the integration branch
`w/2.4/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.4`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #b4befb9f37b64da70d6e5a9efc4f478063c522cb on the integration branch
`w/2.2/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.2`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #b4befb9f37b64da70d6e5a9efc4f478063c522cb on the integration branch
`w/2.3/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.3`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #b4befb9f37b64da70d6e5a9efc4f478063c522cb on the integration branch
`w/2.4/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.4`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **create_pull_requests**

alexandre-allard commented on eve/main.yml:
```diff
@@ -5,6 +5,205 @@ branches:
+  - ShellCommand: &ssh_ip_setup
+      name: Report IP and install SSH keys
+      command: >
+        ip a &&
```
Not related to the PR, but this command can be improved to only return the IP we want (eth0):
```
ip -f inet addr show eth0 | sed -En 's/^.*inet ([0-9.]+).*$/\1/p'
```
Or if we want all interfaces:
```
ip -f inet addr show | sed -En 's/^.*inet ([0-9.]+).* (.+)$/\2: \1/p'
```

alexandre-allard commented on eve/main.yml:
```diff
@@ -5,6 +5,205 @@ branches:
+      name: Debug step - wait befor allowing resource destruction
+      timeout: 3600
+      command: sleep 3600
+      doStepIf: false
```
I think we should use some property defaulted to False for this doStepIf, so we would be able to trigger debug on build without changing this file and doing a debug commit.

gdemonet committed:
ci: Improve SSH connection info reporting

bert-e commented:
# History mismatch



Merge commit #1183590b4362f6b2c2e09f7fe5a69d88b3b46ff7 on the integration branch
`w/2.1/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #1183590b4362f6b2c2e09f7fe5a69d88b3b46ff7 on the integration branch
`w/2.2/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.2`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #1183590b4362f6b2c2e09f7fe5a69d88b3b46ff7 on the integration branch
`w/2.3/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.3`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #1183590b4362f6b2c2e09f7fe5a69d88b3b46ff7 on the integration branch
`w/2.4/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.4`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #4f65b0b7307cd7b16116ac3f577ad7014586abd3 on the integration branch
`w/2.2/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.2`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #4f65b0b7307cd7b16116ac3f577ad7014586abd3 on the integration branch
`w/2.3/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.3`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #1183590b4362f6b2c2e09f7fe5a69d88b3b46ff7 on the integration branch
`w/2.4/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.4`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #e7282628d43ba3058d71c4e54b7d5dcd7efe2a01 on the integration branch
`w/2.1/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

gdemonet committed:
ci: Improve debug steps enabling

Previously, we had a simple `sleep` command, disabled by default.
Enabling it required a dedicated commit, to enable the step in
`eve/main.yml`.

Using this new approach, one can set the build property `debug` on a
manual build trigger from Eve's UI, without requiring the need for any
change in the code itself.

To enable all "debug" steps, set the property to "all".
To enable a subset of these steps, set the property to a tilde-separated
list of step names (e.g. "single-node~multiple-nodes").

Note the use of tilde is forced on us by limitations from Eve accepted
values in build properties (commas, pounds, or spaces, are forbidden).
Previously, we had a simple `sleep` command, disabled by default.
Enabling it required a dedicated commit, to enable the step in
`eve/main.yml`.

Using this new approach, one can set the build property `debug` on a
manual build trigger from Eve's UI, without requiring the need for any
change in the code itself.

To enable all "debug" steps, set the property to "all".
To enable a subset of these steps, set the property to a tilde-separated
list of step names (e.g. "single-node~multiple-nodes").

Note the use of tilde is forced on us by limitations from Eve accepted
values in build properties (commas, pounds, or spaces, are forbidden).

bert-e commented:
# History mismatch



Merge commit #e7282628d43ba3058d71c4e54b7d5dcd7efe2a01 on the integration branch
`w/2.2/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.2`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #e7282628d43ba3058d71c4e54b7d5dcd7efe2a01 on the integration branch
`w/2.3/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.3`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #e7282628d43ba3058d71c4e54b7d5dcd7efe2a01 on the integration branch
`w/2.4/bugfix/increase-timeout-for-large-artifacts` is merging a branch which is neither the current
branch `bugfix/increase-timeout-for-large-artifacts` nor the development branch
`development/2.4`.

It is likely due to a rebase of the branch `bugfix/increase-timeout-for-large-artifacts` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **create_pull_requests**

alexandre-allard commented on eve/main.yml:
```diff
@@ -5,6 +5,256 @@ branches:
+        Must use either:
+          - "all", to select all debug steps
+          - a single step name
+          - a list of step names, separated by tilde signs "~",
```
Weird choice `~` as separation character between step names.
I would have used commas or semicolons, but ok, it's a detail. :)
Otherwise, I like this debugging system, way more usable than the previous one.

alexandre-allard commented on eve/main.yml:
```diff
@@ -5,6 +5,256 @@ branches:
+        # customize the command behaviour. DURATION default value is 3600
+        # (wait duration, in seconds).
+        STEP_NAME: default
+      timeout: 3600
```
Timeout after 3600, so basically you can't put a `$DURATION` longer than 1 hour. :)
If I'm right, the timeout is triggered if there is no output during the period defined, but the step can be longer if there is something on stdout/err.
So maybe, you can have a loop sleeping for N minutes max (where N is less than global timeout) and printing something between each sleep.

gdemonet replied:
See https://github.com/scality/metalk8s/pull/1946/files/916275ac08f58c8dc937240e6c8975fa4a2c3882#diff-73e74177595a435f88a0701e605cf08eR592-R596

gdemonet commented:
/status

bert-e commented:
# Status





*Status report is not available.*






*The following options are set:* **create_pull_requests**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **create_pull_requests**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/increase-timeout-for-large-artifacts` into
`development/2.0`
* [w/2.1/bugfix/increase-timeout-for-large-artifacts](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/bugfix/increase-timeout-for-large-artifacts) will be merged into `development/2.1` (pull request #1953)
* [w/2.2/bugfix/increase-timeout-for-large-artifacts](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/bugfix/increase-timeout-for-large-artifacts) will be merged into `development/2.2` (pull request #1954)
* [w/2.3/bugfix/increase-timeout-for-large-artifacts](https://github.com/scality/metalk8s/compare/development/2.3...w/2.3/bugfix/increase-timeout-for-large-artifacts) will be merged into `development/2.3` (pull request #1955)
* [w/2.4/bugfix/increase-timeout-for-large-artifacts](https://github.com/scality/metalk8s/compare/development/2.4...w/2.4/bugfix/increase-timeout-for-large-artifacts) will be merged into `development/2.4` (pull request #1968)



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **create_pull_requests**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **create_pull_requests**

gdemonet commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`

* :heavy_check_mark: `development/2.3`

* :heavy_check_mark: `development/2.4`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **create_pull_requests, approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`

* :heavy_check_mark: `development/2.3`

* :heavy_check_mark: `development/2.4`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



Please check the status of the associated issue None.

Goodbye gdemonet.
